src/TrackPanel.cpp

```diff
@@ -754,18 +760,36 @@ void TrackPanel::OnMouseEvent(wxMouseEvent & event)
         if(parent)
         {
             wxMenu contextMenu;
-            contextMenu.Bind(wxEVT_MENU,
-               [this]( wxCommandEvent &event ){
-                  OnContextMenu( /*GetProject(),*/ event ); }
-            );
             
-            contextMenu.Append(kContextMenuItemID_AddMonoTrack, XO("Add mono track").Translation());
-            contextMenu.Append(kContextMenuItemID_AddStereoTrack, XO("Add stereo track").Translation());
-            contextMenu.Append(kContextMenuItemID_AddLabelTrack, XO("Add label track").Translation());
-            contextMenu.AppendSeparator();
-            contextMenu.Append(kContextMenuItemID_ExportProject, XO("Export project\tCtrl+E").Translation());
-            contextMenu.AppendSeparator();
-            contextMenu.Append(kContextMenuItemID_SelectAll, XO("Select All\tCtrl+A").Translation());
+            // Get cursor position area: on track or non track (empty area).
+            const auto track = FindTrack (FindCell(event.m_x, event.m_y).pCell.get());
+            if(track)
+            {
```

Comment:
Did you try it with a Label track?  Those tracks already have their own context menus.

I think these context menus should appear only on Wave tracks.